cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/merger/TypeScriptMerger.java

```diff
@@ -63,107 +86,125 @@ public String getType() {
-                jsScript.eval(engineScope);
-            } catch (ScriptException e) {
-                throw new MergeException(base, "Could not evaluate " + scriptName
-                    + " script on initialization. This is most properly a bug. Please report on Github.", e);
-            }
+            sendRequest(mergeTO, conn, os, osw);
+
+            BufferedReader br = new BufferedReader(new InputStreamReader((conn.getInputStream())));
+
+            LOG.info("Receiving output from Server....");
+            Stream<String> s = br.lines();
+            s.parallel().forEachOrdered((String line) -> {
+                if (line.startsWith("import ") || isExportStatement(line)) {
+                    importsAndExports.append(line);
+                    importsAndExports.append(LINE_SEP);
+                } else {
+                    body.append(line);
+                    body.append(LINE_SEP);
+                }
+            });
 
-            scriptEngines.put(scriptName, jsEngine);
+        } catch (IllegalStateException e) {
+
+            LOG.error("Closing connection on InputReader.", e);
+            request.terminateProcessConnection();
+        } catch (Exception e) {
+            connectionExc.setConnectExceptionMessage("Connection to server failed, attempt number " + 0 + ".");
+            connectionExc.setIOExceptionMessage("IO exception when merging");
+
+            connectionExc.handle(e);
         }
+        return runBeautifierExcludingImports(importsAndExports.toString(), body.toString());
+    }
 
-        Invocable jsInvocable = (Invocable) scriptEngines.get(scriptName);
-        try {
-            return (String) executable.exec(jsInvocable);
-        } catch (NoSuchMethodException e) {
-            throw new MergeException(base,
-                "Invalid API of " + scriptName + " script used. This is most properly a bug. Please report on Github.",
-                e);
-        } catch (ScriptException e) {
-            throw new MergeException(base, "Execution of the script " + scriptName + " raised an error.", e);
+    /**
+     * Sends a transfer object to the server, by using http connection
+     * @param dataTO
+     *            Data to transfer, it should be a serializable class
+     * @param conn
+     *            {@link HttpURLConnection} to the server, containing also the URL
+     * @param os
+     *            {@link OutputStream} that will be opened for sending data
+     * @param osw
+     *            {@link OutputStreamWriter} used for writing the data to be sent
+     * @throws IOException
+     *             when connection to the server failed
+     * @throws JsonGenerationException
+     *             When generating the JSON from the transfer object failed
+     * @throws JsonMappingException
+     *             When mapping the JSON from the transfer object failed
+     */
+    private void sendRequest(Object dataTO, HttpURLConnection conn, OutputStream os, OutputStreamWriter osw)
```

Comment:
dataTo

Reply:
btw. maybe even code to be shared in cobigen API? as every plugin would need this code right?

Reply:
I agree, you will find it implemented [here](https://github.com/devonfw/tools-cobigen/pull/840/files#diff-3c4d8d9150d5465bd1104dbbf44849b4R310)